src/AEG/CutNode.ts

```diff
@@ -236,6 +236,65 @@ export class CutNode {
         return false;
     }
 
+    /**
+     * Determines whether or not this cut and the given cut are equivalent.
+     * @param otherCut The other cut that is being compared to
+     * @returns Whether or not these cuts have all of the same nodes
+     */
+    public equals(otherCut: CutNode): boolean {
+        //Two empty cuts will always be equal
+        if (this.internalChildren.length === 0 && otherCut.children.length === 0) {
+            return true;
+        }
+
+        //These two cuts have different amount of children they cannot be the same cut
+        if (this.internalChildren.length !== otherCut.children.length) {
+            return false;
+        }
+
+        const foundChildren: boolean[] = [];
+        for (let i = 0; i < this.internalChildren.length; i++) {
+            foundChildren.push(false);
+        }
+
+        let j: number;
+        for (let i = 0; i < this.internalChildren.length; i++) {
+            j = 0;
+            //Compare this nodes current child to all of the children to see if it exists in it
+            while (!foundChildren[i] && j < otherCut.children.length) {
```

Comment:
On second thought, why is this a while loop instead of a for loop? It appears we have all the elements of a for loop, a counter and a condition.